src/libraries/System.Net.Http/src/System/Net/Http/HttpMessageInvoker.cs

```diff
@@ -53,7 +74,40 @@ public HttpMessageInvoker(HttpMessageHandler handler, bool disposeHandler)
             }
             CheckDisposed();
 
+            if (HttpTelemetry.Log.IsEnabled() && !request.WasSentByHttpClient() && request.RequestUri != null)
+            {
+                return SendAsyncWithTelemetry(_handler, request, cancellationToken);
+            }
+
             return _handler.SendAsync(request, cancellationToken);
+
+            static async Task<HttpResponseMessage> SendAsyncWithTelemetry(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                HttpTelemetry.Log.RequestStart(request);
+
+                try
+                {
+                    return await handler.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                }
+                catch when (LogRequestFailed(telemetryStarted: true))
+                {
+                    // Unreachable as LogRequestFailed will return false
+                    throw;
+                }
+                finally
+                {
+                    HttpTelemetry.Log.RequestStop();
+                }
+            }
+        }
+
+        internal static bool LogRequestFailed(bool telemetryStarted)
+        {
+            if (HttpTelemetry.Log.IsEnabled() && telemetryStarted)
+            {
+                HttpTelemetry.Log.RequestFailed();
+            }
```

Comment:
Is this correct? Wouldn't you want to log `RequestFailed` if `telemetryStarted`, so there is a stop event or counter decrement etc.? 

Or does it literally not matter if logging is already disabled?

Reply:
The scenario of logging being re-enabled after being disabled is not considered - https://github.com/dotnet/runtime/pull/37619#discussion_r444580315

The pattern of checking `*Telemetry.Log.IsEnabled()` is recognized by the linker to remove code when not needed.